irohad/multi_sig_transactions/state/impl/mst_state.cpp

```diff
@@ -104,6 +104,21 @@ namespace iroha {
     return out;
   }
 
+  void MstState::iterateBatches(std::function<void(const DataType &)> visitor) const {
+    std::for_each(internal_state_.cbegin(), internal_state_.cend(), visitor);
+  }
+
+  void MstState::iterateTransactions(
+      std::function<
```

Comment:
std::function is not considered as a good solution for non-tests code due to its performance drawbacks.